analyzer/spec/functional/surface_spec_spec.rb

```diff
@@ -7,48 +7,77 @@ module Interpreter
+        it { expect {
+            spec.interpret("position :c1, :c2, face: 100, dir: :front")
+          }.not_to raise_error }
 
-          before(:each) do
-            spec.interpret('atoms c1: C%d, c2: C%d')
-          end
-
-          it { expect { spec.interpret('position :c1, :c2, face: 100') }.
-            to raise_error(*incomplete) }
+        it { expect {
+            spec.interpret("no-position :c1, :c2, face: 100, dir: :front")
+          }.to raise_error(*syntax_error('non_position.impossible')) }
 
-          it { expect { spec.interpret('position :c1, :c2, dir: :front') }.
-            to raise_error(*incomplete) }
+        describe 'when position already present' do
+          before do
+            spec.interpret("position :c3, :c2, face: 100, dir: :front")
+          end
 
           it { expect {
-              spec.interpret('position :c1, :c2, face: 100, dir: :front')
+              spec.interpret("no-position :c1, :c2, face: 100, dir: :front")
             }.not_to raise_error }
         end
 
-        describe 'only one atom has lattice' do
-          before(:each) do
-            spec.interpret('atoms c1: C%d, c2: C')
+        describe 'when only bond presented' do
+          before do
+            spec.interpret("bond :c3, :c2, face: 100, dir: :front")
           end
 
           it { expect {
-              spec.interpret('position :c1, :c2, face: 100, dir: :front')
-            }.to raise_error(*syntax_error(
-              'surface_spec.undefined_relation', relation: position_100_front))
-          }
+              spec.interpret("no-position :c1, :c2, face: 100, dir: :front")
+            }.to raise_error(*syntax_error('non_position.impossible')) }
         end
+      end
 
-        describe 'duplication' do
-          let(:spec) { Interpreter::SurfaceSpec.new(bridge_base) }
+      %w(position no-position).each do |relation|
+        describe "##{relation}" do
+          describe 'both atoms has lattice' do
+            let(:incomplete) { syntax_error('position.incomplete') }
 
-          it { expect {
-              spec.interpret('position :cl, :cr, face: 100, dir: :front')
-            }.to raise_error(*syntax_warning(
-              'position.duplicate', face: 100, dir: :front)) }
+            it { expect { spec.interpret("#{relation} :c1, :c2, face: 100") }.
+              to raise_error(*incomplete) }
+
+            it { expect { spec.interpret("#{relation} :c1, :c2, dir: :front") }.
+              to raise_error(*incomplete) }
+          end
+
+          describe 'only one atom has lattice' do
+            before(:each) do
+              spec.interpret('atoms c1: C%d, c2: C')
+            end
+
+            let(:undefined_relation) do
+              (relation =~ /^no-/) ? non_position_100_front : position_100_front
+            end
+
+            it { expect {
+                spec.interpret("#{relation} :c1, :c2, face: 100, dir: :front")
+              }.to raise_error(*syntax_error(
```

Comment:
`end` at 68, 14 is not aligned with `expect {` at 66, 17 or `it { expect {` at 66, 12